server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -929,7 +929,11 @@ public IndexResult index(Index index) throws IOException {
             }
         } catch (RuntimeException | IOException e) {
             try {
-                maybeFailEngine("index", e);
+                if (treatDocumentFailureAsTragicError(index)) {
+                    failEngine("index", e);
```

Comment:
can we add more info about document into the "reason" string?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/43523/commits/8725216747348bf96a3ac628bf0d89212fca2723

Reply:
I meant some info about the document itself, i.e. the id of the document (This could help in figuring out why the given failure happened)

Reply:
I added more info in https://github.com/elastic/elasticsearch/pull/43523/commits/c6325268009cc909bf1fadd92ca020f60a42efff.